src/RTT_Format_Reader/RTT_Mesh_Reader.cc

```diff
@@ -25,70 +27,150 @@ void RTT_Mesh_Reader::transform2CGNS(void)
+        else if (cell_name == "tetrahedron")
+            cell_def = Element_Definition::TETRA_4;
+        else if (cell_name == "quad_pyr")
+            cell_def = Element_Definition::PYRA_5;
+        else if (cell_name == "tri_prism")
+            cell_def = Element_Definition::PENTA_6;
+        else if (cell_name == "hexahedron")
+            cell_def = Element_Definition::HEXA_8;
+        else
+        {
+            if (rttMesh->get_dims_ndim() == 2)
+                cell_def = Element_Definition::POLYGON;
+            else if (rttMesh->get_dims_ndim() == 3)
+                cell_def = Element_Definition::POLYHEDRON;
+        }
+
+        unique_element_types.push_back(cell_def);
 
         if (cell_def == Element_Definition::POLYGON)
         {
-            if (rttMesh->get_dims_ndim() == 2)
+            Insist(rttMesh->get_dims_ndim() == 2,
+                   "Polygon cell definition only supported in 2D");
+            rtt_dsxx::SP<CellDef> cell_definition(
+                rttMesh->get_cell_defs_def(cd));
+
+            std::vector<Element_Definition> elem_defs;
+            elem_defs.push_back(Element_Definition(Element_Definition::BAR_2));
+
+            std::vector<int> side_types(cell_definition->get_nsides(), 0);
+
+            cell.reset(new rtt_mesh_element::Element_Definition(
+                cell_definition->get_name(),
+                rttMesh->get_dims_ndim(),
+                cell_definition->get_nnodes(),
+                cell_definition->get_nsides(),
+                elem_defs,
+                side_types,
+                cell_definition->get_all_sides()));
+        }
+        else if (cell_def == Element_Definition::POLYHEDRON)
+        {
+            Insist(rttMesh->get_dims_ndim() == 3,
+                   "Polyhedron cell definition only supported in 3D");
+            rtt_dsxx::SP<CellDef> cell_definition(
+                rttMesh->get_cell_defs_def(cd));
+
+            std::vector<int> side_types;
+            std::vector<Element_Definition> elem_defs;
+
+            // check to see if the QUAD_9 element is present in the
+            // cell_definitions
+            // and check for QUAD_4 in the event that a QUAD_9 is in fact
+            // present
+            bool have_quad9(false);
+            bool have_quad4(false);
+            for (unsigned s = 0; s < cell_definition->get_nsides(); ++s)
+            {
+                int side_type(cell_definition->get_side_types(s));
+
+                std::vector<int>::iterator sit(
+                    std::find(side_types.begin(), side_types.end(), side_type));
+                if (sit == side_types.end())
+                {
+                    Element_Definition elem_def(*cell_definitions[side_type]);
+                    if (elem_def.get_type() == Element_Definition::QUAD_4)
+                        have_quad4 = true;
+                    else if (elem_def.get_type() == Element_Definition::QUAD_4)
+                        have_quad9 = true;
+                }
```

Comment:
No it just sets the bool if the element types are present. The result is checked later.